test/loaders.js

```diff
@@ -327,3 +328,115 @@ test('support context in loader', async (t) => {
     }
   })
 })
+
+test('rersolver unknown type', async t => {
+  const app = Fastify()
+
+  const resolvers = {
+    test: 2
+  }
+
+  app.register(GQL, {
+    resolvers
+  })
+
+  try {
+    // needed so that graphql is defined
+    await app.ready()
+    app.graphql('query { test }')
+  } catch (error) {
+    t.equal(error.message, 'Cannot find type test')
+  }
+})
+
+test('minJit is not a number, throw error', async t => {
+  const app = Fastify()
+
+  app.register(GQL, {
+    jit: '0'
+  })
+  const typeError = new Error('the jit option must be a number')
+
+  try {
+    // needed so that graphql is defined
+    await app.ready()
+  } catch (error) {
+    t.deepEqual(error, typeError)
+  }
+})
+
+test('options cache is type = number', async t => {
+  const app = Fastify()
+
+  app.register(GQL, {
+    cache: 256
+  })
+
+  // needed so that graphql is defined
+  await app.ready()
+})
+
+test('options cache is type != number', async t => {
+  const app = Fastify()
+
+  app.register(GQL, {
+    cache: 'cache'
```

Comment:
Can you make the validation of the cache option a bit tighter?

Reply:
what do you have in mind?